Lib/scilab/scirun.swg

```diff
@@ -118,26 +129,44 @@ SwigScilabPtrToObject(void *pvApiCtx, int iVar, void **pObjValue, swig_type_info
     printError(&sciErr, 0);
     return SWIG_ERROR;
   }
-
-  if (iType == sci_pointer) {
-    sciErr = getPointer(pvApiCtx, piAddrVar, pObjValue);
-    if (sciErr.iErr) {
-      printError(&sciErr, 0);
-      return SWIG_ERROR;
-    }
+  if (iType != sci_pointer) {
+    return SWIG_ERROR;
   }
-  else {
+
+  sciErr = getPointer(pvApiCtx, piAddrVar, &sobj);
+  if (sciErr.iErr) {
+    printError(&sciErr, 0);
     return SWIG_ERROR;
   }
 
+  if (sobj) {
+      SwigSciWrapperObj *wrapper = static_cast<SwigSciWrapperObj*>(sobj);
+      swig_cast_info *cast = SWIG_TypeCheck(wrapper->ty->name, descriptor);
+      if (!cast) {
+          return SWIG_ERROR;
+      }
+      int newmemory = 0;
```

Comment:
This mixes declarations and code, which isn't allowed by C90 and is causing a test failure: https://travis-ci.org/swig/swig/jobs/114753648

Just moving this line before the `if` is probably enough.